src/librustc_typeck/check/mod.rs

```diff
@@ -3763,8 +3763,36 @@ pub fn resolve_ty_and_def_ufcs<'a, 'b, 'tcx>(fcx: &FnCtxt<'b, 'tcx>,
                                                         &'a [ast::PathSegment],
                                                         def::Def)>
 {
+
+    // Associated constants can't depend on generic types.
+    fn have_disallowed_generic_consts<'a, 'tcx>(fcx: &FnCtxt<'a, 'tcx>,
+                                                def: def::Def,
+                                                ty: Ty<'tcx>,
+                                                span: Span,
+                                                node_id: ast::NodeId) -> bool {
+        match def {
+            def::DefAssociatedConst(..) => {
+                if ty::type_has_params(ty) || ty::type_has_self(ty) {
```

Comment:
OK, I see, I was a bit confused. As a temporary bypass for an ICE, this change (probably) makes sense. I thought at first you were imposing a more permanent restriction against associated constants whose types depending on type parameters -- and this didn't seem to make any sense to me, as I don't see any fundamental problem there.

We do need to get our story straight about constants that cannot be resolved until trans, clearly -- seems to me that "constant equality" ought to be handled in a similar way to "type equality" in any case (and hence constant evaluation at type-check time is basically a form of normalization). But I have to finish a write-up talking about normalization and how we should consider a (mildly) different approach there too, so I guess it would apply equally. 

*Anyway*...